rb-files/app.rb

```diff
@@ -0,0 +1,113 @@
+require_relative 'book'
+require_relative 'persons'
+require_relative 'students'
+require_relative 'teachers'
+require_relative 'rental'
+
+class App
+  def initialize
+    @books = []
+    @people = []
+    @rentals = []
+  end
+
+  def welcome
+    puts 'Welcome to School Library App!\n'
+  end
+
+  def list_books
+    if @books.empty?
+      puts 'No books found'
+    else
+      @books.each { |book| puts "Title: \"#{book.title}\", Author: #{book.author}" }
+    end
+  end
+
+  def show_person(person)
+    print "[#{person.class}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
+    puts ", Specialization: #{person.specialization}" if person.is_a?(Teacher)
+    puts ", Classroom: #{person.classroom}" if person.instance_of?(Student)
+  end
+
+  def list_people
+    return puts 'No people found' if @people.empty?
+
+    @people.each { |person| puts "[#{person.class}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}" }
+  end
+
+  def create_person
+    print 'Do you want to Create a students (1) or a teacher (2)? [Input the number]: '
+    select_person = gets.chomp.to_i
+    print 'Name: '
+    name = gets.chomp
+    print 'Age: '
+    age = gets.chomp.to_i
+    case select_person
+    when 1
+      print 'Has parent permission? [Y/N]: '
+      permission = gets[0].capitalize
+      permission = permission == 'Y'
+      #   @people << Student.new(nil, age, name, permission)
+      print 'What is the student\'s classroom? '
+      classroom = gets.chomp
+      @people.push(Student.new(age, name, permission, classroom))
+    when 2
+      print 'Specialization: '
+      specialization = gets.chomp
+      @people << Teacher.new(age, name, specialization)
+    end
```

Comment:
- [ ] Creating a Teacher feature is not working as expected. When I enter the specialization for the Teacher, the app crashes with an error shown in the screenshot below. Please fix that so a teacher is created successfully just as the student creation. 👍🏽
- [ ] 
<img width="1171" alt="Screenshot 2023-04-13 at 6 56 49 PM" src="https://user-images.githubusercontent.com/50941074/231857404-bc9cf1bc-9ba7-464b-895b-5be2f5380947.png">

Reply:
- Well resolved @Talha-Hanif5666